File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        res = []
        n = len(arr)
        for index in range(n - 1):
            greatest = arr[index+1]
            for j in range(index + 1, n):
                if arr[j] > greatest:
                    greatest = arr[j]
            arr[index] = greatest
        arr[len(arr)-1] = -1
        return arr

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> freq = new HashMap<>();
        for(int i : nums) {
            freq.put(i, freq.getOrDefault(i, 0) + 1);
        }
        List<Integer> list = new ArrayList<>(freq.values());
        Collections.sort(list);
        List<Integer> lastK = list.subList(list.size() - k, list.size());
        int[] res = new int[k];
        int index = 0;
        for(Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            int value = entry.getValue();
            int key = entry.getKey();
            if(lastK.contains(value) && index < k) {
                res[index] = key;
                index++;
            }
        }
        return res;
    }
}
